pkg/controllers/nodeclass/status/ami.go

```diff
@@ -32,7 +32,7 @@ type AMI struct {
 }
 
 func (a *AMI) Reconcile(ctx context.Context, nodeClass *v1beta1.EC2NodeClass) (reconcile.Result, error) {
-	amis, err := a.amiProvider.Get(ctx, nodeClass, &amifamily.Options{})
+	amis, err := a.amiProvider.List(ctx, nodeClass)
```

Comment:
Totally optional: Check out https://github.com/awslabs/operatorpkg/blob/main/singleton/controller.go to implement this singleton controller.

Reply:
I will include that in a follow-up